src/main/java/org/jahia/modules/dxwebpack/Main.java

```diff
@@ -0,0 +1,171 @@
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Locale;
+import java.util.stream.Collectors;
+
+@Component(service = {javax.servlet.http.HttpServlet.class, javax.servlet.Servlet.class}, property = {"alias=/appshell", "osgi.http.whiteboard.servlet.asyncSupported=true"})
+public class Main extends HttpServlet {
+    private static final String PACKAGE_JSON = "javascript/apps/package.json";
+    private static final String JAHIA = "jahia";
+    private static final String APPS = "apps";
+    private static Logger logger = LoggerFactory.getLogger(Main.class);
+
+    @Override
+    protected void service(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        try {
+            int slashIndex = request.getPathInfo().indexOf('/', 1);
+            String app = slashIndex == -1 ? request.getPathInfo().substring(1) : request.getPathInfo().substring(1, slashIndex);
+
+            JahiaUser currentUser = JCRSessionFactory.getInstance().getCurrentUser();
+            Locale uiLocale;
+            JCRUserNode userNode;
+            if (!JahiaUserManagerService.isGuest(currentUser)) {
+                userNode = JahiaUserManagerService.getInstance().lookupUserByPath(currentUser.getLocalPath());
+                uiLocale = UserPreferencesHelper.getPreferredLocale(userNode, LanguageCodeConverters.resolveLocaleForGuest(request));
+            } else {
+                response.sendError(HttpServletResponse.SC_FORBIDDEN);
+                return;
+            }
+
+            HttpServletRequestWrapper wrapper = new HttpServletRequestWrapper(request) {
+                @Override
+                public String getContextPath() {
+                    return Jahia.getContextPath();
+                }
+            };
+            wrapper.setAttribute("appName", app);
+            setCustomAttributes(currentUser, uiLocale, wrapper);
+
+            List<String> scripts = getApplicationScripts(app, null);
+            wrapper.setAttribute("scripts", "[" + StringUtils.join(scripts, ",") + "]");
+
+            wrapper.getRequestDispatcher("/modules/dx-commons-webpack/root.jsp").include(wrapper, response);
+        } catch (Exception e) {
+            logger.error("Error while dispatching: {}", e.getMessage(), e);
+        }
+    }
+
+    private void setCustomAttributes(JahiaUser currentUser, Locale uiLocale, HttpServletRequestWrapper wrapper) throws JSONException {
+        wrapper.setAttribute("contextPath", Jahia.getContextPath());
+        wrapper.setAttribute("currentUser", currentUser);
+
+        SettingsBean settingsBean = SettingsBean.getInstance();
+
+        JSONObject config = new JSONObject();
+        config.put("environment", settingsBean.getString("jahia.environment", ""));
+        config.put("maxUploadSize", settingsBean.getJahiaFileUploadMaxSize() / (1024 * 1024));
```

Comment:
Magic number

Reply:
do you want a constant ?

Reply:
in my opinion not really magic, i think the context is clear enough and a developer will know what 1024 means when we talk about bytes. 1048576 would have been more magic, though

Reply:
and i hate useless constants which will make the code more unclear, so I won't add one :-)

Reply:
It was only a random message from me to say I review JAVA, no problem with this code ;)